test/e2e/windows/kubelet_stats.go

```diff
@@ -234,6 +234,29 @@ func findWindowsNode(ctx context.Context, f *framework.Framework) (v1.Node, erro
 	return targetNode, nil
 }
 
+// findWindowsNodes finds all Windows nodes that are Ready and Schedulable
+func findWindowsNodes(ctx context.Context, f *framework.Framework) ([]v1.Node, error) {
+	selector := labels.Set{"kubernetes.io/os": "windows"}.AsSelector()
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+
+	if err != nil {
+		return nil, err
+	}
+
+	var targetNodes []v1.Node
+	for _, n := range nodeList.Items {
+		if e2enode.IsNodeReady(&n) && e2enode.IsNodeSchedulable(&n) {
+			targetNodes = append(targetNodes, n)
+		}
+	}
+
+	if len(targetNodes) == 0 {
+		e2eskipper.Skipf("Could not find and ready and schedulable Windows nodes")
```

Comment:
I don't think we want to skip in this method, and if we do please indicate that will happen in the name and function comment.
I think it would be better to return the list of nodes and let each test case decide if it should skip or not.